extensions/gitpod-remote/src/machineSettings.ts

```diff
@@ -0,0 +1,70 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { parse as parseJson, modify as modifyJson, applyEdits as applyEditsJson } from 'jsonc-parser';
+import { GitpodExtensionContext } from 'gitpod-shared';
+import { PortsStatus } from '@gitpod/supervisor-api-grpc/lib/status_pb';
+
+export async function configureMachineSettings(context: GitpodExtensionContext, supervisorPortList: PortsStatus.AsObject[]) {
+	const extRemoteLogsUri: vscode.Uri = context.logUri;
+	const remoteUserDataPath = path.posix.dirname(path.posix.dirname(path.posix.dirname(path.posix.dirname(context.logUri.path))));
+	const machineSettingsResource = extRemoteLogsUri.with({ path: path.posix.join(remoteUserDataPath, 'Machine', 'settings.json') });
```

Comment:
Emmmmm, it can be
<img width="674" alt="image" src="https://user-images.githubusercontent.com/20944364/192610837-3bea3504-d5cd-4cc8-b35d-9dc98d513331.png">

Reply:
`context.logUri` also contains the extension id as a folder inside exthost folder